conn.go

```diff
@@ -25,25 +24,24 @@ func beginHandlerOpen(ch chan error, removeHandlers func()) func(ev *js.Object)
 
 // closeError allows a CloseEvent to be used as an error.
 type closeError struct {
-	*dom.CloseEvent
+	*js.Object
```

Comment:
This is purely an optional style suggestion - it doesn't change behavior - but I think it'd be nicer.

Since you're already embedding the `*js.Object` into this struct, you might as well add the JS fields here too, and then you can use them in `Error()` method more nicely.

What I mean is that `closeError` can look more like the [`*dom.CloseEvent`](https://gotools.org/honnef.co/go/js/dom#CloseEvent) which it used to embed before this PR:

```Go
// closeError allows a CloseEvent to be used as an error.
type closeError struct {
	*js.Object
	Code     int    `js:"code"`
	Reason   string `js:"reason"`
	WasClean bool   `js:"wasClean"`
}
```

Then you can just do `if e.WasClean {`, `fmt.Sprintf("CloseEvent: (%s) (%d) %s", cleanStmt, e.Code, e.Reason)`, etc.

What do you think of that idea?